### ISSUE TITLE
Auto-detect java installations

### ISSUE THREAD

**Issue by flexagoon** [ATLauncher/ATLauncher#519]
**Is your feature request related to a problem? Please describe.**
It's sometimes hard to find where java is installed

**Describe the solution you'd like**
Auto-detect existing java installations and provide a dropdown like MultiMC does

**Comment by flexagoon** [comment]
Oh wait, looks like I forgot to reply lol.

@RyanTheAllmighty the thing is that I use NixOS Linux so Java is always located in different places inside of `/nix/store`

**Comment by RyanTheAllmighty** [comment]
Can you please provide some sample paths to where the java executable is (For example `/some/path/to/java/11/jdk/bin/java`) and then I can add to the paths. I tried to get an install of NixOS going to find for myself, but it's way more involved than a usual Linux install.

**Comment by flexagoon** [comment]
> and then I can add to the paths

Not really. Java is always located in, for example `/nix/store/<hash>-jdk11/bin/java`. The think is that `<hash>` is just a lot of random letters and numbers and it changes with every system update. So hardcoding the path is not possible here. I'm not sure how other devs do it, but MultiMC somehow actually manages to find all of the existing java installations

**Comment by RyanTheAllmighty** [comment]
Yeah we use a glob to look for any bin/java files anywhere inside a directory. My concern will be how big the /nix/store folder is to do that lookup since it seems that directory holds every package.

From what I've seen in MultiMC it just does similar directory based lookups as we do with essentially the same paths. Does it manage to pick up all the installs on your system or just one maybe, the default one?

**Comment by flexagoon** [comment]
I unfortunately can't send a screenshot now, but it detects all of the installations. The default one which is in the PATH just shows up as `openjdk11` or something like that, while the rest of installations are shown as a full path (`/nix/store/...`)

**Comment by flexagoon** [comment]
![image](https://user-images.githubusercontent.com/66178592/143503794-61d1911e-69f5-4272-ba51-0606b342da2b.png)
@RyanTheAllmighty here's how MultiMC detects my Java installations.

**Comment by RyanTheAllmighty** [comment]
What's your `JAVA_HOME` env variable set to? Which one of those (if any)?

**Comment by flexagoon** [comment]
My `JAVA_HOME` is set to the second one in the list (Java 8). That's probably why it's also detected as `java` and not only as the full `/nix/store` path

**Comment by LuminarLeaf** [comment]
If possible I think something like Prism Launcher's implementation would be a lot more suited for NixOS

Prism Implementation:
https://github.com/PrismLauncher/PrismLauncher/blob/48235f5b71be19bb905a12cf45f4abe5499f85ea/launcher/java/JavaUtils.cpp#L158

How its used in nix while packaging:
https://github.com/NixOS/nixpkgs/blob/5e4fbfb6b3de1aa2872b76d49fafc942626e2add/pkgs/by-name/pr/prismlauncher/package.nix#L116
